components/MintControls/index.tsx

```diff
@@ -1,40 +1,41 @@
 import { parseEther } from '@ethersproject/units'
 import { Valenftines } from 'abis/types'
 import ValenftinesAbi from 'abis/Valenftines.json'
+import Button from 'components/Button'
 import { ContractTransaction } from 'ethers'
-import { useAtomValue } from 'jotai/utils'
+import { useAtomValue, useUpdateAtom } from 'jotai/utils'
 import { earlyMintProofForAddress, isEarlyMintEligble } from 'lib/earlyMint'
 import { mintCostETH } from 'lib/mintCost'
-import { isEarlyMint, isMintLive } from 'lib/mintTiming'
-import Link from 'next/link'
-import { mintAtom, PAGE_STATE } from 'pages'
+import { isEarlyMintLive, isPublicMintLive } from 'lib/mintTiming'
+import { connectModalOpenAtom, mintAtom, PAGE_STATE } from 'pages'
 import { useCallback, useEffect, useMemo, useState } from 'react'
-import styles from 'styles/Mint.module.scss'
-import { SupportedChainId } from 'types'
+import { ALL_SUPPORTED_CHAIN_IDS, SupportedChainId } from 'types'
 import { useAccount, useContract, useNetwork, useSigner } from 'wagmi'
 
 import { VALENFTINES_ADDRESS } from '../../constants'
+import styles from './MintControls.module.scss'
 
 interface MintControlsProps {
   pageState: PAGE_STATE
   setPageState: (p: PAGE_STATE) => void
 }
 
 export default function MintControls({ pageState, setPageState }: MintControlsProps) {
+  const setConnectModalOpen = useUpdateAtom(connectModalOpenAtom)
   const [hasEarlyMinted, setHasEarlyMinted] = useState<boolean>()
   const [{ data: accountData }] = useAccount({
     fetchEns: false,
   })
-  const isEarlyMinter = useMemo(() => {
-    return isEarlyMint() && accountData && isEarlyMintEligble(accountData.address)
-  }, [accountData])
-
-  const mintLive = useMemo(() => {
-    return isMintLive()
-  }, [])
+  const isEarlyMinter = useMemo(() => accountData && isEarlyMintEligble(accountData.address), [accountData])
+  const publicMintLive = useMemo(isPublicMintLive, [])
+  const earlyMintLive = useMemo(isEarlyMintLive, [])
```

Comment:
separated this out into address identity and time window instead of both in one var

Reply:
Does isPublicMiktLive not need parens?

Reply:
No since those are both functions you can pass them and usememo will call internally